### PR TITLE
Update rippled beast

### PR DESCRIPTION
**Pull request by miguelportilla**
This update addresses the slow OS X debug syncs. The new beast multi-buffer class offers a significant performance increase with peer and websocket connections.

### CODE REVIEW

**Comment by MarkusTeufelberger** [comment]
I think I duplicated a "static" line by accident (line 223 in 09e8313bdcea6cf4652a586970ff750aaf26bada) which probably shouldn't be there, at least looking at it it seems out of place.

Also the "if" could be moved within the function body in case you want less duplicated code:

````
static
void
info_handler (SSL const* ssl, int event, int)
{
#if OPENSSL_VERSION_NUMBER < 0x10100000L
    if ((ssl->s3) && (event & SSL_CB_HANDSHAKE_START))
    {
        if (disallowRenegotiation (ssl, SSL_in_before (ssl)))
            ssl->s3->flags |= SSL3_FLAGS_NO_RENEGOTIATE_CIPHERS;
    }
#else
// empty, flag removed in OpenSSL 1.1
#endif
}
````

**Comment by HowardHinnant** [comment]
After correcting the compile-time errors (nothing serious, obsolete `using namespace` and a missing `#include`), I'm getting unittest failures:
````
ripple.app.Manifest load/store
#4 failed: Manifest_test.cpp(299)
#12 failed: Manifest_test.cpp(330)
#13 failed: Manifest_test.cpp(332)
````
and:
````
ripple.app.Manifest validator token
#1 failed: Manifest_test.cpp(454)
Assertion failed: (this->is_initialized()), function operator->, file /Users/howardhinnant/Development/boost_1_64_0/boost/optional/optional.hpp, line 1116.
Abort trap: 6
````

**Comment by vinniefalco** [comment]
Do not attempt to rebase the merge commit or git will blow up. This pull request has to get merged first. Also, I suggest that at least 1 person carefully review this commit "Changes for Beast version 55" as it changes logic. A bug here can prevent the server from handling connections.

**Comment by vinniefalco** [comment]
@HowardHinnant Where is the missing include? Is it in a Beast source file?

**Comment by HowardHinnant** [comment]
````
--- a/src/ripple/overlay/impl/OverlayImpl.h
+++ b/src/ripple/overlay/impl/OverlayImpl.h
@@ -24,6 +24,7 @@
 #include <ripple/core/Job.h>
 #include <ripple/overlay/Overlay.h>
 #include <ripple/overlay/impl/TrafficCount.h>
+#include <ripple/overlay/impl/TMHello.h>
 #include <ripple/server/Handoff.h>
````

File: src/ripple/rpc/impl/ServerHandlerImp.cpp [hunk 1]
```diff
@@ -738,24 +728,23 @@ ServerHandlerImp::statusResponse(
...
-        msg.reason = "Internal Server Error";
+        msg.result(beast::http::status::internal_server_error);
         msg.body = "<HTML><BODY>Server cannot accept clients: " +
             reason + "</BODY></HTML>";
     }
     msg.version = request.version;
-    msg.fields.insert("Server", BuildInfo::getFullVersionString());
-    msg.fields.insert("Content-Type", "text/html");
-    prepare(msg, beast::http::connection::close);
+    msg.insert("Server", BuildInfo::getFullVersionString());
```
Comment (mellery451) [inline src/ripple/rpc/impl/ServerHandlerImp.cpp]: looks like beast now has an enum for standard header fields (convertible to string) - should we consider switching to the enum in places where we build requests or responses like this?

Comment (vinniefalco) [inline src/ripple/rpc/impl/ServerHandlerImp.cpp]: `beast::http::field::server` not a bad idea!

**Comment by vinniefalco** [comment]
@mellery451 There's no reason to turn it on in rippled. But, it shouldn't hurt anymore, because the sequences produced by `multi_buffer` should be logarithmic on input size rather than linear (!)

File: bin/ci/ubuntu/build-and-test.sh [hunk 2]
```diff
@@ -66,11 +66,13 @@ fi
...
 gdb -return-child-result -quiet -batch \
     -ex "set env MALLOC_CHECK_=3" \
+    -ex "set demangle-style none" \
     -ex "set print thread-events off" \
+    -ex "file $APP_PATH" \
+    -ex "set args $APP_ARGS" \
     -ex run \
     -ex "thread apply all backtrace full" \
     -ex "quit" \
-    --args $APP_PATH $APP_ARGS
```
Comment (vinniefalco) [inline bin/ci/ubuntu/build-and-test.sh]: Why did you change this?

Comment (miguelportilla) [inline bin/ci/ubuntu/build-and-test.sh]: On Travis, the following two symbols in this build cause `c++filt` and `gdb` to crash. Setting the demangle style to none prevents the crash.
```_ZSt7forwardIRKZN6ripple10BaseWSPeerINS0_16ServerHandlerImpENS0_9SSLWSPeerIS2_EEE3runEvEUlRT_E_EOS6_RNSt16remove_referenceIS6_E4typeE
_ZSt7forwardIRKZN6ripple10BaseWSPeerINS0_16ServerHandlerImpENS0_9SSLWSPeerIS2_EEE3runEvEUlRT_E_EOS6_RNSt16remove_referenceIS6_E4typeE```

Comment (vinniefalco) [inline bin/ci/ubuntu/build-and-test.sh]: is Beast using c++filt on Travis?

Comment (mellery451) [inline bin/ci/ubuntu/build-and-test.sh]: not really - we run the `rippled` unit tests under `gdb` in travis, presumably to gather a stack track if there is a crash.  `gdb` tries to demangle symbols when it loads the binary, and it's crashing on these symbols now. This is a `gdb` bug -- an alternative solution would be to update `gdb` on the Travis setup, assuming this is indeed fixed in a current release. I think we wanted to first confirm that demangling was the source of the CI crash with this change, then we can decide whether to try upgrading the tools or just skip the demangling like this.

Comment (vinniefalco) [inline bin/ci/ubuntu/build-and-test.sh]: I like the idea of running with a debugger on Travis and getting info. Its a good approach.

**Comment by nbougalis** [comment]
@MarkusTeufelberger: you didn't duplicate it, but you did end up removing `static` from another function. It's not a huge deal. We can fix that.

**Comment by MarkusTeufelberger** [comment]
Thanks, was more of a quick hack to get rippled compiling on my systems again anyways. :-)

**Comment by miguelportilla** [comment]
@nbougalis @MarkusTeufelberger My last commit fixed the missing static and optimized the define as Markus suggested.

**Comment by vinniefalco** [comment]
A bit late but "Changes for Beast version 59" should read "Changes for Beast version 79"
